### PR TITLE
chore(agw): 'nose' is replaced by 'pytest' for unittests

### PR DESCRIPTION
**Pull request by mpfirrmann**
Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

## Summary

The `nose` package for testing is not maintained anymore and is replaced by `pytest`. In this PR, only the python unit tests for the AGW (https://docs.magmacore.org/docs/lte/dev_unit_testing) are migrated. The migration of the S1AP integration tests is tackled in a follow-up PR, #13060.

## Test Plan

```vagrant@magma-dev-focal:~/magma/lte/gateway$ make test_python```
```vagrant@magma-dev-focal:~/magma/lte/gateway$ make test_python_service MAGMA_SERVICE=<service_name>```
(the `UT_PATH` does need a bug fix, #13054)

Outputs:
```
------------------------ generated xml file: /var/tmp/test_results/tests_lte.xml ------------------------

----------- coverage: platform linux, python 3.8.5-final-0 -----------
Coverage XML written to file /var/tmp/codecovs/cover_lte.xml

======================== 434 passed, 4 skipped, 118 warnings in 99.23s (0:01:39) ========================
...
------------------------ generated xml file: /var/tmp/test_results/sudo_lte.xml -------------------------

----------- coverage: platform linux, python 3.8.5-final-0 -----------
Coverage XML written to file /var/tmp/codecovs/cover_sudo_lte.xml

======================== 154 passed, 3 skipped, 14 warnings in 527.74s (0:08:47) ========================
...
----------------------- generated xml file: /var/tmp/test_results/tests_orc8r.xml -----------------------

----------- coverage: platform linux, python 3.8.5-final-0 -----------
Coverage XML written to file /var/tmp/codecovs/cover_orc8r.xml

================================== 142 passed, 123 warnings in 54.91s ===================================
```
vs the existing:
```
Ran 438 tests in 87.672s
...
Ran 152 tests in 544.081s
...
Ran 142 tests in 49.833s
```
The `nose` output does not count the skipped tests separately.

## Additional Information

There are quite a lot of deprecation warnings popping up using `pytest`. These are out of scope here, but should be addressed in the future.

- [ ] This change is backwards-breaking

### CODE REVIEW

**Comment by sebathomas** [comment]
Can you add the test output to the test plan? Ideally containing the counts before and after the change to make sure all tests are still executed.

File: orc8r/gateway/python/python.mk [hunk 1]
```diff
@@ -113,10 +113,10 @@ $(PROTO_LIST): %_protos:
 	@echo "Generating python code for $* .proto files"
 	@mkdir -p $(PYTHON_BUILD)/gen
 	@echo "$(PYTHON_BUILD)/gen" > $(SITE_PACKAGES_DIR)/magma_gen.pth
-	$(BIN)/python $(SRC)/protos/gen_protos.py $(SRC)/$*/protos/ $(MAGMA_ROOT),$(MAGMA_ROOT)/orc8r/protos/prometheus $(SRC) $(PYTHON_BUILD)/gen/
+	$(BIN)/python$(PYTHON_VERSION) $(SRC)/protos/gen_protos.py $(SRC)/$*/protos/ $(MAGMA_ROOT),$(MAGMA_ROOT)/orc8r/protos/prometheus $(SRC) $(PYTHON_BUILD)/gen/
```
Comment (sebathomas) [inline orc8r/gateway/python/python.mk:116]: Are the `$(PYTHON_VERSION)` changes necessary? I think `python` should already link to `python3.8`?

**Review by sebathomas — APPROVED** [review]
:+1:

File: orc8r/gateway/python/python.mk (hunk 1 above)
Comment (mpfirrmann) [reply orc8r/gateway/python/python.mk:116]:
```
vagrant@magma-dev-focal:~/build/python/bin$ python
Python 2.7.18 (default, Mar  8 2021, 13:02:45) 
[GCC 9.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> 
```
That's why I included the `$(PYTHON_VERSION)`.

Comment (sebathomas) [reply orc8r/gateway/python/python.mk:116]: So you think the tests were using Python 2 before? Would be surprising. Or do they run in a virtualenv?

It's probably safe to change in any case.

Comment (mpfirrmann) [reply orc8r/gateway/python/python.mk:116]: I don't know to be honest. But I didn't want to risk it.

**Review by Neudrino — APPROVED** [review]

**Pull request merged**